### ISSUE TITLE
Si-sortable may not be able to parse a more complex ng-repeat string

### ISSUE THREAD

**Issue by rmckeel** [simplicitylabs/si-table#29]
In the code:

```
<tr si-sortable="single" ng-repeat="alert in alertFiltered = (myData | toArray:false)">...</tr>
```

toArray is a filter

Executing this, I get:

Unexpected end of expression: $parent.currentList = alertFiltered = (myData

I have a partial recreation of this issue here, without siTable yet integrated:
http://jsfiddle.net/ds027cq1/1/

The code functions as expected, however the console message above is triggered.

Examining si-table.js a bit, it appears that the plugin parses ng-repeat as a string, and perhaps there is something about this syntactically correct ng-repeat line that is not parsed properly by si-table?

UPDATE: It looks like line ~94 in si-table.js where:
    var collection = matches[2]...

Is returning this error, as it improperly parses the pipe before toArray, and then can't scope.$watchCollection on that partial string that includes the parentheses.

Commenting out if (attrs.ngRepeat) ... from line ~90-100 makes my application work again.  I am uncertain the function / value of watching a string like this then setting offset to 0.
   $parent.currentList = alertFiltered = myData

Thoughts on how to fix / make more tolerant to these cases?

Thanks,

  Ryan

**Comment by mikberg** [comment]
Could you create a fiddle which demonstrates the problem, with siTable integrated?

@limonte do you have any suggestions?

**Comment by rmckeel** [comment]
@mikberg Yes, I will aim to create a demo by end of week.

**Comment by rmckeel** [comment]
I'm having a hard time re-creating the problem.  http://jsfiddle.net/ds027cq1/13/ is fairly close, but no bug message seen yet.

In my code, I have something like this as matches[2] from the 'var collection = matches[2]'... from si-table.js ~ line 94.

"$parent.currentList = $parent.alertFiltered = (status.data.alert | toArray:false ) | orderBy:sortingParams.sortArray | siPagination:paginationParams"

Then, when it tries to split that by "|" character, it gets the wrong pipe.

I will keep trying to reproduce, but perhaps that is more to go on, or more to try fiddling with.

Thanks ~

**Comment by rmckeel** [comment]
Hello, I suppose since I can't reproduce it in a boiled down form, and I can't share the original code, and it is no longer that important to me, I'll close out the issue.  I have resolved the issue by removing < 10 lines of code in si-table.js ~ line 94 that I did not need in my application.

Perhaps this ticket will be useful if someone else has a similar issue and can provide more detail.

Thanks!

  Ryan
